src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -402,21 +412,19 @@ bool ValidateInstructions(ReadOnlySpan<byte> code, in EofHeader header)
                 return false;
             }
 
-            foreach (int rjumpdest in rjumpdests)
+
+            BitArray result = rjumpdestsMask.And(immediatesMask);
+            for (int i = 0; i < result.Length; i++)
             {
-                foreach (var range in immediates)
+                if (result.Get(i))
                 {
-                    if (range.Includes(rjumpdest))
+                    if (Logger.IsTrace)
                     {
-                        if (Logger.IsTrace)
-                        {
-                            Logger.Trace($"EIP-4200 : Static Relative Jump destination {rjumpdest} is an Invalid, falls within {range}");
-                        }
-                        return false;
+                        Logger.Trace($"EIP-4200 : Static Relative Jump with invalid destination");
                     }
+                    return false;
```

Comment:
This loop can be vectorized with `Bytes.CountBits`

Reply:
Even better, we could avoid allocation of `BitArray result` with doing custom loop doing Vectorized `And` and `PopCount` by ourselves in loop spin.

Reply:
I avoided BitArray completely because it's a memory hog apparently, and our implementation of bitmap in the Analyzers is far better : 
* BenchmarkDotNet=v0.13.2, OS=Windows 11 (10.0.22621.963)
* Intel Core i5-8265U CPU 1.60GHz (Whiskey Lake), 1 CPU, 8 logical and 4 physical cores
* .NET SDK=7.0.100
  [Host]     : .NET 6.0.11 (6.0.1122.52304), X64 RyuJIT AVX2
  DefaultJob : .NET 6.0.11 (6.0.1122.52304), X64 RyuJIT AVX2
* Benchmark code : https://github.com/DemuirGos/Benchmarks/tree/EofBenchmarks

|                        Method |         Mean |      Error |     StdDev |    Gen0 | Allocated |
|------------------------------ |-------------:|-----------:|-----------:|--------:|----------:|
|       Bitarray_Valid_Bytecode | 88,553.94 ns | 912.070 ns | 761.620 ns | 48.7061 |  152845 B |
|   BinarySearch_Valid_Bytecode | 36,552.01 ns | 498.805 ns | 442.178 ns |  2.6855 |    8468 B |
|         Bitmap_Valid_Bytecode | 32,655.47 ns | 354.905 ns | 331.978 ns |  0.0610 |     344 B |
|       NaiveWay_Valid_Bytecode | 34,253.70 ns | 656.176 ns | 673.845 ns |  2.6855 |    8508 B |
|     Bitarray_Invalid_Bytecode |    113.56 ns |   1.895 ns |   1.680 ns |  0.1633 |     512 B |
| BinarySearch_Invalid_Bytecode |     66.46 ns |   1.337 ns |   1.373 ns |  0.0255 |      80 B |
|       Bitmap_Invalid_Bytecode |     77.54 ns |   1.565 ns |   1.387 ns |  0.0867 |     272 B |
|     NaiveWay_Invalid_Bytecode |     65.25 ns |   0.854 ns |   0.799 ns |  0.0255 |      80 B |